app/controllers/main/volunteerRegisterEvents.py

```diff
@@ -0,0 +1,67 @@
+from app.models.event import Event
+from app.models.user import User
+from app.models.eventParticipant import EventParticipant
+from app.controllers.admin import admin_bp
+from app.controllers.events import events_bp
+from app.logic.userRsvpForEvent import userRsvpForEvent, unattendedRequiredEvents
+from flask import flash, request, redirect, url_for, g
+
+
+
+@admin_bp.route('/rsvpForEvent', methods=['POST'])
+def volunteerRegister():
+    """
+    This function selects the user ID and event ID and registers the user
+    for the event they have clicked register for.
+    """
+    eventData = request.form
+    userId = User.get(User.username == g.current_user)
+    eventId = eventData['eventId']
+    program = eventData['programId']
+    isEligible = userRsvpForEvent(userId, eventId)
+    listOfRequirements = unattendedRequiredEvents(program, userId)
+
+
+    if not isEligible: # if they are banned
+        flash(f"Cannot RSVP: {userId.firstName} is banned")
+
+    elif len(listOfRequirements) > 0:
+        if len(listOfRequirements) >= 3:
+            reqListToString = ', '.join(listOfRequirements[:-1])
+            reqListToString += ' or ' + listOfRequirements[-1]
+            flash(f"{userId.firstName} {userId.lastName} successfully registered. Warning! You have not attended the following required trainings: {reqListToString}")
+
+        elif len(listOfRequirements) == 2:
+            reqListToString = ' or '.join(listOfRequirements)
+            flash(f"{userId.firstName} {userId.lastName} successfully registered. Warning! You have not attended the following required trainings: {reqListToString}")
+
+
+        else:
+            reqListToString = str(listOfRequirements[0])
+            reqListToString = 'Berea Buddies'
+            flash(f"{userId.firstName} {userId.lastName} successfully registered. Warning! You have not attended the following required trainings: {reqListToString}")
+
+    #if they are eligible
+    else:
+        RSVPupdate =(EventParticipant.update({EventParticipant.rsvp: True})

```

Comment:
weren't they already registered above in userRsvpForEvent?

Reply:
Yup